File: leetcode/python/0058_length_of_last_word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        # start from the end of s towards the front using i:
        # counting = False initially
        # if counting == False and s[i] is not a space: counting = True, res++
        # elif counting == True and s[i] is not a space: res++
        # elif counting == True and s[i] is a space: break
        # return res
        # edge cases: "   ", "a", "    aaa", "a  "

        res, counting = 0, False
        for i in range(len(s) - 1, -1, -1):
            if s[i] != " ":
                if counting == False:
                    counting = True
                    res += 1
                elif counting == True:
                    res += 1
            elif counting == True and s[i] == " ":
                break
        
        return res

File: leetcode/java/0058_length_of_last_word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int res = 0;
        boolean counting = false;

        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) != ' ') {
                if (counting == false) {
                    counting = true;
                }
                if (counting == true) res++;
            } else if ((s.charAt(i) == ' ' || i == 0) && counting == true) {
                return res;
            }
        }

        return res;
    }
}